pkg/binlog-filter/filter.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb-tools/pkg/table-rule-selector"
+)
+
+// ActionType is how to handle mathed items
+type ActionType string
+
+//  show Actions that how to handle rule
+const (
+	Skip ActionType = "skip"
+	Do   ActionType = "Do"
+)
+
+// EventType is DML/DDL Event type
+type EventType string
+
+// show DML/DDL Events
+const (
+	InsertEevent EventType = "insert"
+	UpdateEvent  EventType = "update"
+	DeleteEvent  EventType = "delete"
+
+	CreateDatabase EventType = "create database"
+	DropDatabase   EventType = "drop database"
+	CreateTable    EventType = "create table"
+	DropTable      EventType = "drop table"
+	TruncateTable  EventType = "truncate table"
+	RenameTable    EventType = "rename table"
+	CreateIndex    EventType = "create index"
+	DropIndex      EventType = "drop index"
+	AlertTable     EventType = "alter table"
+	// if need, add more	AlertTableOption     = "alert table option"
+
+	NullEvent EventType = ""
+)
+
+// BinlogEventRule is a rule to filters binlog events
+type BinlogEventRule struct {
+	SchemaPattern string         `json:"schema-pattern" toml:"schema-pattern"`
+	TablePattern  string         `json:"table-pattern" toml:"table-pattern"`
+	DMLEevent     []EventType    `json:"dml" toml:"dml"`
+	DDLEvent      []EventType    `json:"ddl" toml:"ddl"`
```

Comment:
seems not verify a event config is a DMLEvent or DDLEvent (eg. user can config a `DeleteEvent` in `DDLEvent`)? how about combine them to a single `BinlogEvent`?

Reply:
I plan to check them later, I leave a comment above `Valid()`